### ISSUE TITLE
[Bug] Killing a NevergradSearch run and then resuming fails with KeyError

### ISSUE THREAD

**Issue by froody** [ray-project/ray#18863]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Tune

### What happened + What you expected to happen

Running a simple example with NevergradSearch and then pressing ^C before it ends, and then resuming results in the following crash:

```    self.searcher.on_trial_complete(
  File ".../site-packages/ray/tune/suggest/nevergrad.py", line 273, in on_trial_complete
    self._live_trial_mapping.pop(trial_id)
KeyError: 'abb24ae8'
```

### Reproduction script

```
% python test.py
... # wait for a few trials to complete
^C
% python test.py --resume
... # should crash
```

```import ray
import numpy as np
import time
from ray.tune.suggest.nevergrad import NevergradSearch
import argparse
import nevergrad as ng


def train_function(config, reporter, extra=None):
    print(f"config = {config}")
    time.sleep(1)
    reporter(acc=np.random.rand(), done=True)

def main():
    parser = argparse.ArgumentParser("Hyperparameter search")
    parser.add_argument("--resume", action="store_true", default=False)

    ray.init(address="auto")

    args = parser.parse_args()

    func = train_function
    analysis = ray.tune.run(
        func,
        search_alg=NevergradSearch(optimizer=ng.optimizers.OnePlusOne),
        metric="acc",
        mode="max",
        config={
            "label_smoothing": ray.tune.uniform(0, 1),
        },
        num_samples=100,
        name="test_run",
        resume=args.resume,
    )
    print("best config: ", analysis.get_best_config(metric="acc", mode="max"))


if __name__ == "__main__":
    main()
```

### Anything else

Crashes every time.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by froody** [comment]
Apparently this is fixed on master as of yesterday, when is the next release?

**Comment by krfricke** [comment]
The next release (1.7.0) is currently running final release tests, but this fix will probably only be included in 1.8.0 (in ~1 month)
